File: c++/26_Remove_Duplicates_from_Sorted_ArrayII.cpp
/*************************************************************************
	> File Name: 26_Remove_Duplicates_from_Sorted_ArrayII.cpp
	> Author: Pelhans 
	> Mail: me@pelhans.com
	> Created Time: 2018年03月13日 星期二 13时29分51秒
 ************************************************************************/

#include<iostream>
using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() <= 2) return nums.size();
        
        int count = 1;
        for (int i = 2; i < nums.size(); i++){
            if (!(nums[count-1] == nums[count] && nums[count] == nums[i])){
                nums[++count] = nums[i];
            }
        }
        return count + 1 ;
    }
};

File: python/26_Remove_Duplicates_from_Sorted_ArrayII.py
#!/usr/bin/env python
# coding=utf-8

class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) <=2:
            return len(nums)
        
        count = 1
        for i in range(2, len(nums)):
            if not (nums[count-1] == nums[count] and nums[count] == nums[i]):
                count = count + 1
                nums[count] = nums[i]
        return count + 1
